sql/migrate/dir.go

```diff
@@ -59,6 +53,50 @@ type (
 		// StmtDecls returns the set of SQL statements this file holds alongside its preceding comments.
 		StmtDecls() ([]*Stmt, error)
 	}
+
+	// Formatter wraps the Format method.
+	Formatter interface {
+		// Format formats the given Plan into one or more migration files.
+		Format(*Plan) ([]File, error)
+	}
+)
+
+type (
+	// CheckpointDir wraps the functionality used to interact
+	// with a migration directory that support checkpoints.
+	CheckpointDir interface {
+		// WriteCheckpoint writes the given checkpoint file to the migration directory.
+		WriteCheckpoint(name, tag string, content []byte) error
+
+		// CheckpointFiles returns a set of checkpoint files stored in this Dir,
+		// ordered by name.
+		CheckpointFiles() ([]File, error)
+
+		// FilesFromCheckpoint returns the files to be executed on a database from
+		// the given checkpoint file, including it. An ErrCheckpointNotFound if the
+		// checkpoint file is not found in the directory.
+		FilesFromCheckpoint(string) ([]File, error)
```

Comment:
The input is the full name of the file or just the version?

Reply:
Currently, the file name.